plugin/gostring/gostring.go

```diff
@@ -338,29 +341,34 @@ func (p *gostring) Generate(file *generator.FileDescriptor) {
 	p.Out()
 	p.P(`}`)
 
-	p.P(`func extensionToGoString`, p.localName, `(m `, protoPkg.Use(), `.Message) string {`)
-	p.In()
-	p.P(`e := `, protoPkg.Use(), `.GetUnsafeExtensionsMap(m)`)
-	p.P(`if e == nil { return "nil" }`)
-	p.P(`s := "proto.NewUnsafeXXX_InternalExtensions(map[int32]proto.Extension{"`)
-	p.P(`keys := make([]int, 0, len(e))`)
-	p.P(`for k := range e {`)
-	p.In()
-	p.P(`keys = append(keys, int(k))`)
-	p.Out()
-	p.P(`}`)
-	p.P(sortPkg.Use(), `.Ints(keys)`)
-	p.P(`ss := []string{}`)
-	p.P(`for _, k := range keys {`)
-	p.In()
-	p.P(`ss = append(ss, `, strconvPkg.Use(), `.Itoa(k) + ": " + e[int32(k)].GoString())`)
-	p.Out()
-	p.P(`}`)
-	p.P(`s+=`, stringsPkg.Use(), `.Join(ss, ",") + "})"`)
-	p.P(`return s`)
-	p.Out()
-	p.P(`}`)
-
+	if extensionToGoStringUsed {
+		if !gogoproto.ImportsGoGoProto(file.FileDescriptorProto) {
+			p.Error(fmt.Errorf("Please import gogoproto in '%v' in order to use extensions.", p.localName))
```

Comment:
fmt.Fprintf(os.Stderr, "The GoString plugin for messages with extensions requires importing gogoprotobuf.  Please see file %s", file.Name)
os.Exit(1)